cmd/osctl/cmd/cluster.go

```diff
@@ -100,15 +99,32 @@ func create() (err error) {
 		return err
 	}
 
-	// Generate all PKI and tokens required by Talos.
+	// Validate CIDR range and allocate IPs
+	fmt.Println("validating CIDR and reserving master IPs")
 
-	fmt.Println("generating PKI and tokens")
+	masterIP, _, err := net.ParseCIDR(networkCIDR)
+	if err != nil {
+		return fmt.Errorf("error validating cidr block: %w", err)
+	}
+
+	// Set starting ip at 2nd ip in range, ex: 192.168.0.2
+	masterIP = masterIP.To4()
+	masterIP[3] += 2
```

Comment:
We have https://github.com/talos-systems/talos/blob/master/pkg/net/net.go#L52 to do this for us. Has all kinds of checks built into it.

Reply:
updated with using the nth function. thx for the pointer to that. also updated the names of the containers that get created to prepend the clustername. one step closer to doing multi-clusters on a single node.